### ISSUE TITLE
Add VST plugins support

### ISSUE THREAD

**Issue by chris-zen** [chris-zen/hero-studio#27]
Supporting device chains in tracks for VST plugins will enable the use of effects and instruments. This task will take care of making sure that the studio processing logic knows how to host them and process MIDI and audio for them.
The initial implementation can be single threaded (all the plugins rendered in sequence), but it would be nice also to start figuring out:
1) How to process plugins in parallel with a thread pool taking care of their ownership
2) How to deal with crashes without the DAW dying (super stretch goal) For example Bitwig creates independent processes per plugin, which would require some kind of IPC mechanism, which would go through the OS and might break the rule of lock-free/no-allocations/no-OS-calls.
